### ISSUE TITLE
Placeholder in Select2 Search Input

### ISSUE THREAD

**Issue by luke-** [humhub/humhub#5718]
Add "search..." placeholder into the text box
![image](https://user-images.githubusercontent.com/4736168/169772473-413040df-3ac4-4366-a8ac-c675243f7da6.png)

Would be good if we could integrate such a placeholder text into the various Select2 DropDowns.

See: https://github.com/humhub/team_tasks/issues/224

**Comment by yurabakhtin** [comment]
@luke- I see on your screenshot the field Country looks like Select2 dropdown with input field, but I see it as simple html select element on `develop` branch:

![country_field](https://user-images.githubusercontent.com/6995524/228305530-78365d33-8290-47a4-91cc-da4992f96670.png)

Should I modify the field into Select2 Dropdown firstly and then implement the placeholder for the field?

**Comment by luke-** [comment]
@yurabakhtin this is strange. can you please check why in `develop` the `country` field is no longer a select2. If there is no good reason there, please convert that back to a Select2.

**Comment by yurabakhtin** [comment]
@luke- Sorry, it was my bug from here https://github.com/humhub/humhub/pull/6020/files#diff-f0dddc2afb07bab05bf804dea03a3f7e2ce5c6b2a244e7e772af4b9a8412eae6R107, I forgot to merge options from `CountrySelect` because it is extended from `Select` with additional options.
Fixed here https://github.com/humhub/humhub/pull/6208/commits/0e26915f13ea0e3d70198f8a99ec0fbb495464bc#diff-f0dddc2afb07bab05bf804dea03a3f7e2ce5c6b2a244e7e772af4b9a8412eae6R107.

**Comment by yurabakhtin** [comment]
@luke- Placeholder for the Country field has been added here https://github.com/humhub/humhub/pull/6208/commits/ad0ce2d5b523fb92adac0c47f34d74563726b78f#diff-4bc3a1ae05bf3a9510b1b2e24e7556f4f77a90b00fbe8ac341302caa3a0b8b32R103-R104.

Please note if a placeholder is enabled for Select2 widget then original empty option "Please select:" is hidden, and the placeholder is displayed only at the place:

![country_placeholder](https://user-images.githubusercontent.com/6995524/228443653-7d8b3b93-ca1b-4865-a125-8ff16596a524.png)

In order to select an empty value user should click here:

![empty_select](https://user-images.githubusercontent.com/6995524/228444056-439e4267-796d-4c6f-9b73-9131745feb92.png)

**Comment by luke-** [comment]
Sorry if it was not clear, I'm  referring to the following input box. 
![image](https://user-images.githubusercontent.com/4736168/228476792-4f2f33ad-210c-410f-9f79-bc768bdc2db3.png)

**Comment by yurabakhtin** [comment]
@luke- Ok, I have implemented this for the field "Language" in commit https://github.com/humhub/humhub/pull/6208/commits/f20b724fafbfd5c29a60a303a745d3f0ecc78d5d:

![language](https://user-images.githubusercontent.com/6995524/228508688-9f6b66f2-e205-4be4-a586-fb69740412af.png)

Please note by default the plugin "Select2" doesn't put a placeholder in the place, see above my screenshot for default placeholder location, so I used an extenssion from here https://github.com/andreivictor/select2-searchInputPlaceholder in order to have the placeholder in the requested place as on your screenshots.

---

Should I revert the placeholder from the Country field, or is it ok to have it as this? :

![country](https://user-images.githubusercontent.com/6995524/228509360-a3fb5602-996d-4215-bab9-75c842d36d69.png)

---

Currently the placeholders are implemented only for 2 fields "Country" and "Language", if we need this for all select2 fields I could suggest to implament new field like:

```php
$form->field($model, 'language')->select2($languages);
```

instead of current code in many places:

```php
$form->field($model, 'language')->dropDownList($languages, [
    'data-ui-select2' => '',
    'data-search-input-placeholder' => Yii::t('UserModule.base', 'Search...'),
]);
```

**Comment by luke-** [comment]
@yurabakhtin Please see my comments in the PR regarding the search field. Regarding the "Value" Placeholder, can you please describe the differences a bit more in detail. 

If you think it is useful, we can create our own Select2 Form widget.

**Comment by yurabakhtin** [comment]
@luke- 

> Regarding the "Value" Placeholder, can you please describe the differences a bit more in detail.

1. In order to initialize the JS plugin Select2 we need to add attribute `data-ui-select2`:

![select2](https://user-images.githubusercontent.com/6995524/228741802-4f377c5e-7a13-4597-8bff-efa8be3e6a89.png)

2. Option `placeholder` of the Select2 does this:

![select2-placeholder](https://user-images.githubusercontent.com/6995524/228742022-6b7bab3c-0d70-4653-bfb9-4ddceebb6fd0.png)

3. In order to allow to select an empty option we can use option `allow-clear`:

![select2-placeholder-allow-clear](https://user-images.githubusercontent.com/6995524/228742274-10830bbb-5092-4e31-9cf1-edd8afbb796a.png)

4. But in order to keep a placeholder as your screenshot we should use external plugin, because Select2 plugin doesn't have it:

![select2-search-input-placeholder](https://user-images.githubusercontent.com/6995524/228742429-782179ee-03ec-4d75-9923-306aef8d1658.png)

---

> If you think it is useful, we can create our own Select2 Form widget.

Yes, I think it will be useful in order to don't write everytime at least `data-ui-select2` and `data-search-input-placeholder` because you wrote to keep it for all Select2 by default.

**Comment by yurabakhtin** [comment]
@luke- Commit https://github.com/humhub/humhub/pull/6208/commits/e417a91699d80648ad1862ef4de65f6c4350f139 after chat discussion.
